tests/unit/plugins/test_npm_plugin.py

```diff
@@ -241,7 +248,13 @@ def test_get_build_commands_false(self, part_info, new_dir):
         plugin = NpmPlugin(properties=properties, part_info=part_info)
 
         assert plugin.get_build_commands() == [
-            'npm install -g --prefix "${CRAFT_PART_INSTALL}" $(npm pack . | tail -1)',
+            'NPM_VERSION="$(npm --version)"\n'
+            "# use the new-style install command if npm >= 10.0.0\n"
+            "if ((${NPM_VERSION%%.*}>=10)); then\n"
+            '    npm install -g --prefix "${CRAFT_PART_INSTALL}" --install-links "${PWD}"\n'
+            "else\n"
+            '    npm install -g --prefix "${CRAFT_PART_INSTALL}" "$(npm pack . | tail -1)"\n'
+            "fi\n",
         ]
 
     def test_get_build_commands_include_node_true(self, part_info, mocker, new_dir):
```

Comment:
Could you add unit tests for the logic and regex in `_get_best_node_version()`?  Tests for the 3 runtime and value errors in that function would be a bonus.

You could add a `pytest.mark.parameterize` to test some of the common versions you listed (`20.12.2`, `20.12`, `20`, `lts/iron`, `node`) here.  Or a separate test that calls `_get_best_node_version()`, if that's easier.

Reply:
Done